### PR TITLE
Pass pointer events to surface layers

### PR DESCRIPTION
**Pull request by SirCmpwn**

### CODE REVIEW

File: sway/input/cursor.c [hunk 1]
```diff
@@ -20,9 +22,35 @@ static void cursor_update_position(struct sway_cursor *cursor) {
...
+		struct wlr_box box = {
+			.x = sway_layer->geo.x,
+			.y = sway_layer->geo.y,
+			.width = wlr_surface->current->width,
+			.height = wlr_surface->current->height,
+		};
+		// TODO: Test popups/subsurfaces
+		if (wlr_box_contains_point(&box, ox, oy) &&
+				pixman_region32_contains_point(
+					&wlr_surface->current->input, _sx, _sy, NULL)) {
```
Comment (emersion) [inline sway/input/cursor.c]: We'll probably want to wrap this into a wlr function

Comment (SirCmpwn) [reply sway/input/cursor.c]: meh

Comment (SirCmpwn) [reply sway/input/cursor.c]: alright you convinced me

**Comment by SirCmpwn** [comment]
Will switch to wlr_surface_point_accepts_input later so I can avoid some annoying rebasing

**Pull request merged**